### DISCUSSION TITLE
Guantanamo Issues

### DISCUSSION THREAD

**Discussion by TecnoTalarn** [foldynl/QLog#371] [General]
I found an error in the identification of the prefix KG4, which should be Guantanamo Bay, but in some cases it is U.S.A.
According to some discussion groups, it seems that only the callsigns KG4xx are really from Guantanamo, while those with one or three letters in the suffix are "United States".

You should confirm this end and program the code to avoid confusion.
The text I found reads:
_**"The ARRL DXCC prefix list shows KG4 as Guantanamo Bay but as you correctly state only KG4 plus two letters is Guantanamo Bay and KG4 plus either one or three characters should be USA"**_

here: https://forum.log4om.com/viewtopic.php?t=397

**Comment by foldynl** [comment]
this kind of issue has to be reported to the [cty.dat ](https://www.country-files.com) maintainers - prefix database used by QLog. 

If this database does not reflect the correct prefix assignment, then it is necessary to solve it at the application level. However, seems to me as a dirty solution. The clean solution is that cty file should contain a list of all exceptions for this prefix, as it is for all other countries. From developer perspective, it is not possible to monitor exceptions at the level of all national authorities around the world.

**Comment by TecnoTalarn** [comment]
OK, I'll look into talking to them, but it seems strange that they don't know. In clublog, for example, the 3 letter callsigns appear as USA, and in some other programs as well. It will be that they do it at the application level and not the database that you comment on.
Thanks for the information.
let's see what they say to me.

**Comment by foldynl** [comment]
The problem seems to be more complicated. I don't know if it's realistic, but theoretically there can be a variant of the callsign KH6xxx/KG4. Should QLog return it is United State or Guantanamo? More about the callsign variants #227

**Reply by foldynl** [reply]
any idea how to solve it?

**Comment by TecnoTalarn** [comment]
One possibility could be that the callsign start with KG4 or callsign/KG4 or KG4/callsign, the software looks at the locator, if it exists in the profile of hamqth or qrz, in order to decide if it is in the USA or in Guantanamo. If the correspondent reports the locator correctly it should be easy to fix

**Reply by TecnoTalarn** [reply]
Anyway, if the code is /KG4 or KG4/ it will always be Guantanamo... right?

**Reply by foldynl** [reply]
estimating the country according to the locator is a logical step, but it is not reliable. You don't always have a locator. If you are offline, it is also a problem to obtain the loc from QRZ or HamQTH. This approach doesn't seem very deterministic to me. 

If callsign will be /KG4 or KG4/ then it will be always Guantanamo now.

Maybe, it is possible to use [Big CTY](https://www.country-files.com/big-cty/) file as a solution. But I don't see many exceptions for the KG4 callsigns there.

**Comment by TecnoTalarn** [comment]
There are certainly very few exceptions with Guantanamo.
99% of the KG4 are from the USA, and those who are really Guantanamo, are reflected in the large database of the CTY.
Maybe it would be better if the software considered KG4 as Usa, except for /KG4 or KG4/ and it would focus 99% of the cases, leaving only a few erroneous exceptions.
That would be much better than what happens now that all KG4s are considered Guantanamo, and at least in my case, none are.
Really, the CTY people should consider this in order to avoid errors, because now their classification is wrong in most cases.

**Comment by foldynl** [comment]
that makes sense. I will try to look at the BIG CTY file

**Comment by foldynl** [comment]
The fix is a part of 0.37.0
